resteasy/src/main/java/com/linecorp/armeria/server/resteasy/ResteasyAsynchronousResponseImpl.java

```diff
@@ -0,0 +1,245 @@
+        }
+    }
+
+    @Override
+    public boolean resume(Throwable ex) {
+        synchronized (responseLock) {
+            if (done) {
+                return false;
+            }
+            if (cancelled) {
+                return false;
+            }
+            done = true;
+            return internalResume(ex, this::onResponseCompletion);
+        }
+    }
+
+    @Override
+    public boolean cancel() {
+        synchronized (responseLock) {
+            if (cancelled) {
+                return true;
+            }
+            if (done) {
+                return false;
+            }
+            done = true;
+            cancelled = true;
+            return internalResume(Response.status(Response.Status.SERVICE_UNAVAILABLE).build(),
+                                  this::onResponseCompletion);
+        }
+    }
+
+    @Override
+    public boolean cancel(int retryAfter) {
+        synchronized (responseLock) {
+            if (cancelled) {
+                return true;
+            }
+            if (done) {
+                return false;
+            }
+            done = true;
+            cancelled = true;
+            return internalResume(Response.status(Response.Status.SERVICE_UNAVAILABLE)
+                                          .header(HttpHeaders.RETRY_AFTER, retryAfter).build(),
+                                  this::onResponseCompletion);
+        }
+    }
+
+    @Override
+    public boolean cancel(Date retryAfter) {
+        synchronized (responseLock) {
+            if (cancelled) {
+                return true;
+            }
+            if (done) {
+                return false;
+            }
+            done = true;
+            cancelled = true;
+            return internalResume(Response.status(Response.Status.SERVICE_UNAVAILABLE)
+                                          .header(HttpHeaders.RETRY_AFTER, retryAfter).build(),
+                                  this::onResponseCompletion);
+        }
+    }
+
+    @Override
+    public boolean isSuspended() {
+        return !done;
+    }
+
+    @Override
+    public boolean isCancelled() {
+        return cancelled;
+    }
+
+    @Override
+    public boolean isDone() {
+        return done;
+    }
+
+    /**
+     * Set/update the suspend timeout.
+     * <p>
+     * The new suspend timeout values override any timeout value previously specified.
+     * The asynchronous response must be still in a {@link #isSuspended() suspended} state
+     * for this method to succeed.
+     * </p>
+     *
+     * @param time suspend timeout value in the give time {@code unit}. Value lower
+     *             or equal to {@value #NO_TIMEOUT} causes the context to suspend indefinitely.
+     * @param unit suspend timeout value time unit.
+     * @return {@code true} if the suspend time out has been set, returns {@code false} in case
+     *         the request processing is not in the {@link #isSuspended() suspended} state.
+     */
```

Comment:
Javadoc is inherited by default. If the Javadoc is the same as the super method, we don't need to write Javadoc.

Reply:
Right. I've added a note that extending the suspend period by setting a new suspend time-out from inside `TimeoutHandler` is not supported.